hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -158,12 +156,22 @@ func (r *NodePoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			RateLimiter:             workqueue.NewItemExponentialFailureRateLimiter(1*time.Second, 10*time.Second),
 			MaxConcurrentReconciles: 10,
 		}).
-		Build(r)
-	if err != nil {
+		Complete(r); err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
+	if err := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Secret{}).
+		WithOptions(controller.Options{
+			RateLimiter:             workqueue.NewItemExponentialFailureRateLimiter(1*time.Second, 10*time.Second),
+			MaxConcurrentReconciles: 10,
+		}).
+		Complete(&secretJanitor{
+			NodePoolReconciler: r,
```

Comment:
If only Secret metadata is needed, I'm wondering if we can setup the controller-runtime client to cache only the Secret metadata. https://github.com/kubernetes-sigs/controller-runtime/blob/479b723944e34ae42c9911fe01228ff34eb5ca81/pkg/builder/example_test.go#L38